model/project_parser.go

```diff
@@ -653,8 +653,17 @@ func processIntermediateProjectIncludes(ctx context.Context, identifier string,
 		"module":      include.Module,
 	})
 	if include.Module != "" {
-		yaml, err = retrieveFileForModule(ctx, *localOpts, intermediateProject.Modules, include.Module)
-		err = errors.Wrapf(err, "%s: retrieving file for module '%s'", LoadProjectError, include.Module)
+		// Check if the module is a local change passed in through the CLI
+		for _, includeModule := range projectOpts.LocalModuleIncludes {
+			if includeModule.Module == include.Module && includeModule.FileName == include.FileName {
+				yaml = includeModule.FileContent
+				break
+			}
+		}
+		if yaml == nil {
+			yaml, err = retrieveFileForModule(ctx, *localOpts, intermediateProject.Modules, include.Module)
```

Comment:
I think it may be more readable to instead pass LocalModuleIncludes into the localOpts above and move [this](https://github.com/evergreen-ci/evergreen/pull/8005/files#diff-36ad6f3611325fe22762cd371e3130049e163c1ab97d3c55bd011f8827ba0a35R657-R661) logic into `retrieveFileForModule`